o2a/mappers/java_mapper.py

```diff
@@ -0,0 +1,113 @@
+
+from typing import List, Optional, Set
+from xml.etree.ElementTree import Element
+
+from o2a.converter.constants import LIB_FOLDER
+from o2a.converter.relation import Relation
+from o2a.converter.task import Task
+from o2a.mappers.action_mapper import ActionMapper
+from o2a.mappers.extensions.prepare_mapper_extension import PrepareMapperExtension
+from o2a.o2a_libs.property_utils import PropertySet
+from o2a.utils import xml_utils
+from o2a.utils.file_archive_extractors import FileExtractor, ArchiveExtractor
+from o2a.utils.xml_utils import get_tags_el_array_from_text
+
+TAG_MAIN_CLASS = "main-class"
+TAG_JAVA_OPTS = "java-opts"
+TAG_JAVA_OPT = "java-opt"
+TAG_ARG = "arg"
+
+
+class JavaMapper(ActionMapper):
+    """
+    Converts a Java Oozie action node to an Airflow task.
+    """
+
+    def __init__(
+        self,
+        oozie_node: Element,
+        name: str,
+        dag_name: str,
+        props: PropertySet,
+        jar_files: List[str],
+        **kwargs,
+    ):
+        ActionMapper.__init__(
+            self, oozie_node=oozie_node, dag_name=dag_name, name=name, props=props, **kwargs
+        )
+        self.file_extractor = FileExtractor(oozie_node=oozie_node, props=self.props)
+        self.archive_extractor = ArchiveExtractor(oozie_node=oozie_node, props=self.props)
+        self.main_class: Optional[str] = None
+        self.java_opts: Optional[List[str]] = None
+        self.args: Optional[List[str]] = None
+        self.hdfs_files: Optional[List[str]] = None
+        self.hdfs_archives: Optional[List[str]] = None
+        self.prepare_extension: PrepareMapperExtension = PrepareMapperExtension(self)
+        self.jar_files: List[str] = jar_files if jar_files else []
+        self.jar_files_in_hdfs: List[str] = []
+        self._compute_jar_files_in_hdfs()
+
+    def on_parse_node(self):
+        super().on_parse_node()
+        _, self.hdfs_files = self.file_extractor.parse_node()
+        _, self.hdfs_archives = self.archive_extractor.parse_node()
+        self._extract_java_data()
+
+    def to_tasks_and_relations(self):
+        action_task = Task(
+            task_id=self.name,
+            template_name="java.tpl",
+            template_params=dict(
+                props=self.props,
+                hdfs_files=self.hdfs_files,
+                hdfs_archives=self.hdfs_archives,
+                main_class=self.main_class,
+                jar_files_in_hdfs=self.jar_files_in_hdfs,
+                args=self.args,
+            ),
+        )
+        tasks = [action_task]
+        relations: List[Relation] = []
+        prepare_task = self.prepare_extension.get_prepare_task()
+        if prepare_task:
+            tasks, relations = self.prepend_task(prepare_task, tasks, relations)
+        return tasks, relations
+
+    def required_imports(self) -> Set[str]:
+        return {"from airflow.utils import dates", "from airflow.contrib.operators import dataproc_operator"}
+
+    def _compute_jar_files_in_hdfs(self):
```

Comment:
```suggestion
    def _get_jar_files_in_hdfs_full_paths(self):
```

The `compute` bit here sounds a little awkward to me here.

Reply:
👍

Reply:
Fixed in 8d678467f06c67dda8ecad043720608b47c130cf